File: programmers/etc/level2/줄_서는_방법.py
from math import factorial

def solution(n, k):
    answer = []
    num = list(range(1, n+1))
    while n:
        idx = (k-1)//factorial(n-1)
        answer.append(num.pop(idx))
        k = k%factorial(n-1)
        n -= 1
    return answer


n, k = 3, 5
print(solution(n, k)) # [3,1,2]

File: Java/programmers/etc/level2/줄서는방법.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 줄서는방법 {
    public int[] solution(int n, long k) {
        int[] answer = new int[n];
        List<Integer> num = new ArrayList<>();

        long f = 1;
        for (int i=1; i<=n; i++) {
            f *= i;
            num.add(i);
        }

        k--;
        int idx = 0;
        while (idx < n) {
            f /= n-idx;
            answer[idx++] = num.remove((int)(k/f));
            k %= f;
        }
        
        return answer;
    }
}
